File: example/src/main/java/com/example/bundlerexample/Example1Activity.java
package com.example.bundlerexample;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.Log;

import com.hendraanggrian.bundler.Bundler;
import com.hendraanggrian.bundler.annotations.BindExtra;

public class Example1Activity extends BaseActivity {

    @BindExtra("hello") String hello;
    @BindExtra("world") String world;
    @BindExtra("asd") @Nullable String asd;

    @Override
    protected int getContentLayout() {
        return R.layout.activity_example1;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundler.bind(this);
        Log.d("hello", hello);
        Log.d("world", world);
    }
}

File: example/src/com/example/bundler/Example1Activity.kt
package com.example.bundler

import android.os.Bundle
import androidx.annotation.Nullable
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import com.hendraanggrian.bundler.BindExtra
import com.hendraanggrian.bundler.bindExtras
import kotlinx.android.synthetic.main.activity_example1.*

class Example1Activity : AppCompatActivity() {
    @BindExtra lateinit var title: String
    @BindExtra @Nullable @JvmField var subtitle: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_example1)
        setSupportActionBar(toolbar)
        bindExtras()
        titleView.text = title
        if (subtitle != null) {
            subtitleView.isVisible = true
            subtitleView.text = subtitle
        }
    }
}